src/TrackTraversingAlgorithms.cpp

```diff
@@ -29,6 +170,97 @@ void VolumeCalculator::execute() {
 
 
 /**
+ * @brief Constructor for CentroidGenerator calls the TraverseTracks
+ *        constructor and imports refernces to both the FSR volumes and FSR
+ *        locks arrays
+ * @param track_generator The TrackGenerator to pull tracking information from
+ */
+CentroidGenerator::CentroidGenerator(TrackGenerator* track_generator)
+                                  : TraverseTracks(track_generator) {
+
+  _FSR_volumes = track_generator->getFSRVolumes();
+  _FSR_locks = track_generator->getFSRLocks();
+  _quadrature = track_generator->getQuadrature();
+}
+
+
+/**
+ * @brief Calculates the centroid of every FSR
+ * @details On each segment, onTrack(...) calculates the contribution to each
+ *          FSR centroid and saves the centroids in the centroids array
```

Comment:
maybe just "..and saves them in...". There's a lot of `centroids` in this sentence.